bcachefs-tools: add symlink for mount.bcachefs and fix runtime deps

Mic92 opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Mic92 committed:
bcachefs-tools: add symlink for mount.bcachefs and fix runtime deps

Mic92 committed:
nixos/bcachefs: fix boot with systemd enabled initrd

bjornfor commented on nixos/modules/tasks/filesystems/bcachefs.nix:
```diff
@@ -42,7 +42,10 @@ in
     {
-      system.fsPackages = [ pkgs.bcachefs-tools ];
+      # we don't want bachefs in fsPackages for systemd-initrd because we will
+      # to provide the unwrapped version of mount.bcachefs via extraBin into scope.
```
I think this comment needs rewording. (I failed to parse it.)

Mic92 replied:
rewritten.

Mic92 committed:
bcachefs-tools: unstable-2022-09-28 -> unstable-2022-12-29

Diff: https://github.com/koverstreet/bcachefs-tools/compare/99caca2c70f312c4a2504a7e7a9c92a91426d885...42cf74fd1d0ef58927967e6236988e86cfc0d086

Mic92 committed:
linux_testing_bcachefs: 2022-10-31 -> 2022-12-29

Mic92 committed:
nixos/tests/installer: disable zfs for bcachefs tests

we are potentially testing very new kernels here, which are not
supported by zfs yet.

Mic92 committed:
nixos/tests/installer/bcachefs: use ocr to type in password

Mic92 commented:
Turns out that encrypted bcachefs partitions were broken in this kernel release. I fixed it by bumping to the latest commit.
